### DISCUSSION TITLE
Send a poll after deferral.

### DISCUSSION THREAD

**Discussion by Peter-MJ-Parker** [discordjs/discord.js#10367] [Polls]
Hi, I'm wondering why content is required and why the poll will not send if we have deferred the reply. (Code is an example) 
If users are saving questions and answers to database, they may want to await it and defer to save to db before sending or editing the poll. 
I'm guessing the reason is a Poll is not able to be edited after it sent. Is that correct?

```ts
    await interaction.deferReply({ fetchReply: true, ephemeral: true });
    await delay(3);
    await interaction.editReply({
      content: "Test Poll",
      poll: {
        allowMultiselect: false,
        question: { text: "Test Poll" },
        answers: [{ text: "Test Answer1" }, { text: "Test Answer2"}],
        duration: 768
      }
    });
```

**Comment by Qjuh** [comment]
You answered your own question already. Polls can’t be edited (in) so they need to be sent in the initial message. deferReply sends an initial message. So you‘d either need to deferReply ephemerally and editReply that with success or whatnot, then send a followUp containing the poll. Or simply send the poll and save to db asynchronously instead of waiting for the db call to succeed. Or store to db and editReply with a button that will then send the stored poll when clicked.

Polls not being able to be edited in is a Discord API limitation and we can’t do anything about it.
